randovania/resolver/energy_tank_damage_state.py

```diff
@@ -44,8 +43,18 @@ def _maximum_energy(self, resources: ResourceCollection) -> int:
         num_tanks = resources[self._energy_tank]
         return self._starting_energy + (self._energy_per_tank * num_tanks)
 
+    def _duplicate(self) -> Self:
```

Comment:
would it be worth adding this?
```py
def __copy__(self) -> Self:
    return self._duplicate()
```

Reply:
Wouldn't be the first class to have that